graphbolt/src/cuda/neighbor_sampler.cu

```diff
@@ -37,7 +37,7 @@
     defined(__HIP_ARCH_HAS_GLOBAL_INT64_ATOMICS__) && \
     defined(__HIP_ARCH_HAS_SHARED_INT64_ATOMICS__)
 #define __ATOMICS__ 1
-#elif __CUDA_ARCH__ >= 700
+#elif defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 700
```

Comment:
The check for defined isn't necessary. From the C++ standard:

> After all replacements due to macro expansion and evaluations of defined-macro-expressions and has-include expressions have been performed, all remaining identifiers and keywords, except for true and false, are replaced with the pp-number 0

Reply:
I guess it matters with -Wundef, or -Wpedantic, but this project doesn't use those, so I removed various `#if defined()` usages.

Reply:
Oh, I think the style in this project is to still use `#if defined(FOO) || defined(BAR)` if definedness is all you're checking for, and I think maybe that's clearer? But in the case where you're already checking its numeric value the extra `defined(FOO) &&` is mostly noise IMO (given it's not building with `-Wpedantic` and such) and the project currently does it inconsistently, so I've been leaving it off. So instead of `#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 700` you can just write `#if __CUDA_ARCH__ >= 700`